exe/puppet-db.cc

```diff
@@ -107,11 +95,9 @@ main(int argc, char **argv) {
             }
 
             const auto subcommand = vm["subcommand"].as<string>();
-            if ((!(subcommand == "query") &&
-                 !(subcommand == "export") &&
+            if ((!(subcommand == "export") &&
```

Comment:
Not necessarily part of this PR, but I'd suggest 'subcommand != "query"' (assuming the operator works as I expect), and I might break up the logic of detecting that the subcommand is valid from that of vetting the subcommand args (i.e. two separate return blocks).